Dockerfile

```diff
@@ -1,83 +1,19 @@
-  echo '#!/bin/bash\njava -jar SnowBlossomClient_deploy.jar configs/client.conf "$@"' > client.sh && \
-  echo '#!/bin/bash\njava -jar SnowBlossomMiner_deploy.jar configs/miner.conf "$@"' > miner.sh && \
-  echo '#!/bin/bash\njava -jar PoolMiner_deploy.jar configs/pool-miner.conf "$@"' > pool-miner.sh && \
-  echo '#!/bin/bash\njava -Xmx50g -jar Arktika_deploy.jar "$@"' > arktika.sh && \
-  echo '#!/bin/bash\njava -Xmx10g -jar Arktika_deploy.jar "$@"' > arktika-small.sh && \
-  echo '#!/bin/bash\njava -Xmx10g -jar ShackletonExplorer_deploy.jar configs/explorer.conf "$@"' > shackleton-explorer.sh && \
-  chmod +x *.sh && \
-  mkdir -p "logs"
-
-CMD ./node.sh
-
-#to build:  docker build --tag=snowblossom .
-#to run: cp --recursive example/configs configs
-#  mkdir -p node_db wallet snow
-#  chmod go+wt node_db wallet snow
-
-#use --restart=always --detach instead of --rm if you want this to start on boot
-#to run       node: docker run --rm --name=snowblossom-node --volume $PWD/configs:/home/docker/snowblossom/configs --volume $PWD/node_db:/home/docker/snowblossom/node_db --publish 2338:2338 snowblossom
-#to run     client: docker run --rm --name=snowblossom-client --network=host --volume $PWD/configs:/home/docker/snowblossom/configs --volume $PWD/wallet:/home/docker/snowblossom/wallet snowblossom ./client.sh
-#to run      miner: docker run --rm --name=snowblossom-miner --network=host --volume $PWD/configs:/home/docker/snowblossom/configs --volume $PWD/snow:/home/docker/snowblossom/snow --volume $PWD/wallet:/home/docker/snowblossom/wallet snowblossom ./miner.sh
-#to run  poolminer: docker run --rm --name=snowblossom-pool-miner --network=host --volume $PWD/configs:/home/docker/snowblossom/configs --volume $PWD/snow:/home/docker/snowblossom/snow --volume $PWD/wallet:/home/docker/snowblossom/wallet snowblossom ./pool-miner.sh
-#to run    arktika: docker run --rm --name=snowblossom-arktika --network=host --volume $PWD/configs:/home/docker/snowblossom/configs --volume $PWD/snow:/home/docker/snowblossom/snow --volume $PWD/wallet:/home/docker/snowblossom/wallet snowblossom ./arktika-small.sh configs/node1.conf
-#to run shackleton: docker run --rm --name=snowblossom-shackleton-explorer --network=host --volume $PWD/configs:/home/docker/snowblossom/configs --publish 8080:8080 snowblossom ./shackleton-explorer.sh
-#to run   snowfall: n=8; mkdir -p snow/snowblossom.$n && chmod go+wt snow/snowblossom.$n && docker run --rm --name=snowblossom-snowfall --network=host --volume $PWD/configs:/home/docker/snowblossom/configs --volume $PWD/snow:/home/docker/snowblossom/snow --volume $PWD/wallet:/home/docker/snowblossom/wallet snowblossom java -jar SnowFall_deploy.jar snow/snowblossom.$n/snowblossom.$n.snow snowblossom.$n $(dc <<< "2 $n ^ 1024 * p")
-
+ARG BAZEL_VERSION=2.0.0
+ARG BAZEL_OPTIONS
+
+FROM debian:10-slim as build
+ARG BAZEL_VERSION=${BAZEL_VERSION}
+ARG BAZEL_OPTIONS=${BAZEL_OPTIONS}
+COPY . /snowblossom
```

Comment:
Copying the scripts in and running them makes this `Dockerfile` opaque.

It works, though.

One alternate option would be to only copy `.git/`, check out, and do that brunt effort explicitly in the `Dockerfile`. This would shield one from accidentally building uncommitted files locally.

Reply:
I'd like to hear more discussion ideas, but I think I'll move forward for now. :)